routs/soldiers.js

```diff
@@ -0,0 +1,29 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, data, done) {
+
+  const parsedUrlString = req.url.split("/");
+  if (parsedUrlString.length >= 3) {
+    done("invalid url request");
+  }
+  const urlParams = parsedUrlString[2];
+  const parsedUrl = new urlib.URL("http://localhost:3000/" + req.url);
+  if (req.method === "GET") {
+    if (parsedUrlString.length === 3 && /tt[0-9]{7}/.test(urlParams)) {
+      soldiers.findSoldier(urlParams , null, done)
+    } else if (parsedUrl.searchParams.has("name")) {
```

Comment:
What did the task documentation say about other parameters sent as queries? Is it ok to send them? is it ok for you to ignore them?